src/RefractDataStructure.cc

```diff
@@ -917,11 +917,49 @@ namespace drafter {
         }
     };
 
+    static void CheckTypeAttributesClash(const mson::TypeAttributes& attributes,
+                                         const snowcrash::SourceMap<mson::ValueDefinition>& sourceMap,
+                                         ConversionContext& context) {
+
+        if ((attributes & mson::FixedTypeAttribute) != 0 &&
+            (attributes & mson::OptionalTypeAttribute) != 0) {
+
+            context.warn(snowcrash::Warning("cannot use 'fixed' and 'optional' together",
+                                            snowcrash::MSONError, sourceMap.sourceMap));
+        }
+
+        if ((attributes & mson::RequiredTypeAttribute) != 0 &&
+            (attributes & mson::OptionalTypeAttribute) != 0) {
+
+            context.warn(snowcrash::Warning("cannot use 'required' and 'optional' together",
+                                            snowcrash::MSONError, sourceMap.sourceMap));
+        }
+
+        if ((attributes & mson::DefaultTypeAttribute) != 0 &&
+            (attributes & mson::SampleTypeAttribute) != 0) {
+
+            context.warn(snowcrash::Warning("cannot use 'default' and 'sample' together",
+                                            snowcrash::MSONError, sourceMap.sourceMap));
+        }
+
+        if ((attributes & mson::FixedTypeAttribute) != 0 &&
+            (attributes & mson::FixedTypeTypeAttribute) != 0) {
+
+            context.warn(snowcrash::Warning("cannot use 'fixed' and 'fixed-type' together",
+                                            snowcrash::MSONError, sourceMap.sourceMap));
+        }
+    }
+
     template <typename Trait>
     static refract::IElement* MsonMemberToRefract(const typename Trait::InputType& input,
                                                   ConversionContext& context,
                                                   const mson::BaseTypeName nameType,
-                                                  const mson::BaseTypeName defaultNestedType) {
+                                                  const mson::BaseTypeName defaultNestedType,
+                                                  bool checkTypeAttributes = true) {
```

Comment:
We won't get duplicate warnings because of the `context.warn`, having these flags passed around is correct but harder to read, not having it makes it simpler. But leaving to @klokane to make the last call.